tests/automation_framework/tests/work_order_tests/test_submit.py

```diff
@@ -1815,3 +1815,443 @@ def test_work_order_submit_requesterNonce_param_empty(self):
+            constants.wo_submit_output_json_file_name,
+            read_json(request_file))
+
+        result_response = self.test_obj.getresult(
+            self.test_obj.build_request_output['request_obj'])
+        assert (
+                verify_test(
+                    result_response, 0,
+                    self.test_obj.build_request_output['pre_test_output'],
+                    self.test_obj.build_request_output['action_obj'])
+                is TestStep.SUCCESS.value)
+        logger.info('\t\t!!! Test completed !!!\n\n')
+
+
+    @pytest.mark.work_order_submit
+    @pytest.mark.test_workordersubmit_workerEncryptionKey_notdefaulthex
+    @pytest.mark.sdk
+    @pytest.mark.listener
+    def test_workordersubmit_workerEncryptionKey_notdefaulthex(self):
+        request_file = os.path.join(
+            constants.work_order_input_file,
+            "workordersubmit_workerEncryptionKey_notdefaulthex.json")
+
+        err_cd = \
+            self.test_obj.setup_and_build_request_wo_submit(
+                read_json(request_file))
+
+        submit_response = submit_request(
+            self.test_obj.uri_client,
+            self.test_obj.build_request_output['request_obj'],
+            constants.wo_submit_output_json_file_name,
+            read_json(request_file))
+
+        result_response = self.test_obj.getresult(
+            self.test_obj.build_request_output['request_obj'])
+        assert (
+                verify_test(
+                    result_response, 0,
+                    self.test_obj.build_request_output['pre_test_output'],
+                    self.test_obj.build_request_output['action_obj'])
+                is TestStep.SUCCESS.value)
+        logger.info('\t\t!!! Test completed !!!\n\n')
+
+
+    @pytest.mark.work_order_submit
+    @pytest.mark.test_workordersubmit_requesterNonce_notdefaultlength
+    @pytest.mark.listener
+    @pytest.mark.sdk
+    def test_workordersubmit_requesterNonce_notdefaultlength(self):
+        request_file = os.path.join(
+            constants.work_order_input_file,
+            "workordersubmit_requesterNonce_notdefaultlength.json")
+
+        err_cd = \
+            self.test_obj.setup_and_build_request_wo_submit(
+                read_json(request_file))
+
+        submit_response = submit_request(
+            self.test_obj.uri_client,
+            self.test_obj.build_request_output['request_obj'],
+            constants.wo_submit_output_json_file_name,
+            read_json(request_file))
+
+        result_response = self.test_obj.getresult(
+            self.test_obj.build_request_output['request_obj'])
+        assert (
+                verify_test(
+                    result_response, 0,
+                    self.test_obj.build_request_output['pre_test_output'],
+                    self.test_obj.build_request_output['action_obj'])
+                is TestStep.SUCCESS.value)
+        logger.info('\t\t!!! Test completed !!!\n\n')
+
+
+    @pytest.mark.work_order_submit
```

Comment:
Why does the name say encryptedRequestHash when we are not doing anything with that parameter?

Reply:
This test for testing fields in EEA document for section 6.6Work Order Signing, where 2 mechanisms are for checking integrity of work order request. so here I am considering encryptedRequestHash parameter, not requesterSignature, that is other mechanism for checking work order request integrity.